fhir-persistence-schema/docs/DB2SchemaMigration.md

```diff
@@ -0,0 +1,382 @@
+# The IBM FHIR Server - Db2 Schema Migration and Schema Management
+
+This document outlines the migration and management of changes in the IBM FHIR Server schema. These cover the supporting Java code and the supporting SQL. There are three core projects which are referenced: 
+
+| Project | Description |
+|----------|---------------------------------------------|
+| `fhir-model` | The HL7 FHIR model - generated Java code |
+| `fhir-database-utils` | The SQL/database constructs used to create SQL compliant statements |
+| `fhir-persistence-schema` | The code bridges the model and database-utils |
+
+The schema generate the following object types, and require specific managements: 
+
+- DB2 PACKAGE
+- TABLESPACES
+- SCHEMA 
+- GLOBAL VARIABLE 
+- INDEX 
+- PERMISSION 
+- PROCEDURE 
+- SEQUENCE 
+- TABLE
+- TABLE CONSTRAINT 
+
+----------------------------------------------------------------
+# Schema: FHIR_ADMIN
+
+The Schema FHIR_ADMIN is an administrative schema. 
+
+## Managing TABLESPACES
+
+There are multiple TABLESPACES used. For the administrative tablesspaces, the `FHIR_TS` tablespace is created with automatic storage.  The tablespace is managed automatically with a default EXTENTSIZE of 4.  
+
+The `FHIR_TS` tablespace is created one time and should not be changed. 
+
+## Managing SEQUENCES
+
+The FHIR_ADMIN schema has a single sequence - `TENANT_SEQUENCE`. 
+The sequence is created a single time, and is NOT replaced or upgraded at any point in time. 
+
+The actions `fhir-persistence-schema` actions - `Allocate Tenant - AddTenantDAO` and `Add Tenant - AddTenantKeyDAO` depend on `next value for`.  These actions, if the sequence is removed, the `TENANT_SEQUENCE` must be manually recreated starting with an integer higher than the `TENANTS.MT_ID` and `TENANT_KEYS.TENANT_KEY_ID` otherwise a conflict with the PRIMARY_KEY in the `FHIR_ADMIN` tables occurs. 
```

Comment:
```suggestion
The `fhir-persistence-schema` actions - `Allocate Tenant - AddTenantDAO` and `Add Tenant - AddTenantKeyDAO` depend on `next value for`.  If the sequence is removed, the `TENANT_SEQUENCE` must be manually recreated starting with an integer higher than the `TENANTS.MT_ID` and `TENANT_KEYS.TENANT_KEY_ID` otherwise a conflict with the PRIMARY_KEY in the `FHIR_ADMIN` tables occurs. 
```